Porting/add-pod-file

```diff
@@ -0,0 +1,239 @@
+    unless length($section);
+my $section_croak = "$0: Value for command-line switch must be one of @man_section_abbrevs\n";
+$section_croak .= "    Select one initial from:\n$man_sections_str";
+croak $section_croak unless $man_sections_seen{$section};
+
+my $newpodfile = "$stub.pod";
+my $newpodpath = File::Spec->catfile('pod', $newpodfile);
+croak "Unable to locate new file '$newpodpath'" unless -f $newpodpath;
+
+say "Step 1: Basic test of validity of POD in $newpodpath" if $verbose;
+
+my $rv;
+system(qq|$^X cpan/Pod-Checker/podchecker $newpodpath|)
+    and croak "$newpodpath has POD errors";
+system(qq|git add $newpodpath|) and croak "Unable to 'git add'";
+
+# Step 2:  Insert entry for $newpodpath into MANIFEST
+
+say "Step 2: Insert entry for $newpodpath into MANIFEST" if $verbose;
+
+my $manifest = 'MANIFEST';
+open(my $IN, '<', $manifest)
+    or croak "Can't open $manifest for reading";
+my @manifest_orig = <$IN>;
+close($IN) or croak "Can't close $manifest after reading";
+chomp(@manifest_orig);
+
+my (@before_pod, @pod, @after_pod);
+my $seen_pod = 0;
+while (my $l = shift(@manifest_orig)) {
+    if (! $seen_pod and $l !~ m{^pod\/}) {
+        push @before_pod, $l;
+    }
+    elsif ($l =~ m{^pod\/}) {
+        push @pod, $l;
+        $seen_pod++;
+    }
+    else {
+        push @after_pod, $l;
+    }
+}
+
+say "Inserting entry for '$newpodpath' into MANIFEST; text will be '$abstract'" if $verbose;
+my $new_manifest_entry = "$newpodpath\t\t$abstract";
+my @new_pod = sort_manifest(@pod, $new_manifest_entry);
+
+open(my $OUT, '>', $manifest)
+    or croak "Can't open $manifest for writing";
+binmode($OUT);
+say $OUT join("\n", @before_pod, @new_pod, @after_pod);
+close($OUT) or croak "Can't close $manifest after writing";
+
+say "Step 3: Add entry to pod/perl.pod" if $verbose;
+
+# Read the existing pod/perl.pod into memory.
+# Divide it into chunks before the selected section, the head2 of the selected
+# section, the selected section, and what comes after the selected section.
+# Add the stub and abstract for the new .pod file to the end of the selected
+# section.  (Manually reposition to taste.)
+
+my $perlpod = "./pod/perl.pod";
+
+open(my $IN1, '<', $perlpod)
+    or croak "Can't open $perlpod for reading";
+my $perlpod_str;
+{
+    local $/;
+    $perlpod_str = <$IN1>;
+}
+close($IN1) or croak "Can't close $perlpod after reading";
+
+my $section_head = "=head2 $man_sections->{$section}";
+my @chunks = split $section_head, $perlpod_str;
+chomp $chunks[0]; # So we can use 'say' consistently later on
+
+my @balance = split /\n/, $chunks[1];
+shift @balance; # $chunks[1] begins with a newline which we won't need to output
+my (@target_section, @after_section);
+
+my $this = 0;
+for my $l (@balance) {
+    if ($l =~ m/^=(head2|for)/) {
+        push @after_section, $l;
+        $this++;
+    }
+    elsif (! $this) {
+        push @target_section, $l;
+    }
+    else {
+        push @after_section, $l;
+    }
+}
```

Comment:
I found this section confusing, although after a lot of slow re-reading, I think it is correct.  Part of the problem is that I would have tackled this whole problem in a different, but not necessarily better, way.  But in this one section (splitting @balance into "target" and "after") I think I would suggest at minimum two small changes.

1. eliminate `$this`, which is a fairly meaningless name for something with an important meaning
2. replace the if/elsif/else with a (two-clause) if/else

I believe that would replace the code from L201-213 with:

```perl
my $target = \@target_section;
for my $l (@balance) {
    $target = \@after_section if $l =~ m/^=(head2|for)/;
    push @$target, $l;
}
```

The reassignment of $target within the loop will happen many times, but not so many that the performance cost will be noticed, especially given that this program will be run once or twice each year.